src/uct/ib/rc/accel/gga_mlx5.c

```diff
@@ -187,6 +220,210 @@ uct_gga_mlx5_rkey_resolve(uct_ib_mlx5_md_t *md, uct_rkey_t rkey)
 
 static UCS_CLASS_DECLARE_DELETE_FUNC(uct_gga_mlx5_iface_t, uct_iface_t);
 
+static unsigned uct_gga_mlx5_iface_progress(uct_iface_h iface)
+{
+    uct_rc_mlx5_iface_common_t *rc_iface =
+            ucs_derived_of(iface, uct_rc_mlx5_iface_common_t);
+
+    return uct_rc_mlx5_iface_poll_tx(rc_iface, UCT_IB_MLX5_POLL_FLAG_HAS_EP);
+}
+
+static ucs_status_t
+uct_gga_mlx5_iface_event_fd_get(uct_iface_h tl_iface, int *fd_p)
+{
+    uct_rc_mlx5_iface_common_t *iface =
+            ucs_derived_of(tl_iface, uct_rc_mlx5_iface_common_t);
+    uct_ib_mlx5_md_t *md UCS_V_UNUSED = uct_ib_mlx5_iface_md(
+            &iface->super.super);
+
+    ucs_assert(md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_CQ);
+    *fd_p = iface->cq_event_channel->fd;
+    return UCS_OK;
+}
+
+static ucs_status_t
+uct_gga_mlx5_iface_arm(uct_iface_h tl_iface, unsigned events)
+{
+    uct_rc_mlx5_iface_common_t *iface =
+            ucs_derived_of(tl_iface, uct_rc_mlx5_iface_common_t);
+    uct_ib_mlx5_md_t *md UCS_V_UNUSED = uct_ib_mlx5_iface_md(
+            &iface->super.super);
+
+    ucs_assert(md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_CQ);
+    return uct_rc_mlx5_iface_devx_arm(iface, events);
+}
```

Comment:
why can't reuse ucs_status_t uct_rc_mlx5_iface_arm/fd_get?

Reply:
we can but it these wrappers have unrelated branch if `!(md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_CQ)` which I replaced with assertion which should be always `true` for GGA

Reply:
this is slow path, so IMO it's not worth it to have extra functions for gga